core/web/router.go

```diff
@@ -278,9 +279,11 @@ func guiAssetRoutes(box packr.Box, engine *gin.Engine) {
 
 		var is404 bool
 		if matchedBoxPath == "" {
+			isApiRequest, _ := regexp.MatchString(`^/v[0-9]+/.*`, path)
```

Comment:
API requests that have invalid paths fall through to the asset router. Since the asset router now defaults to `index.html`, this was causing API routes to return `200` when they should `404`. 

This change detects if the request was an API request and `404`s if it is.